osquery/sql/virtual_table.cpp

```diff
@@ -582,59 +843,83 @@ static int xFilter(sqlite3_vtab_cursor* pVtabCursor,
 }
+
+using sqlite3_module = struct sqlite3_module;
+
+sqlite3_module* getVirtualTableModule(const std::string& table_name,
+                                      bool extension) {
+  static std::unordered_map<std::string, sqlite3_module> modules;
+  if (modules.find(table_name) != modules.end()) {
+    return &modules[table_name];
+  }
+
+  modules[table_name] = {};
+  modules[table_name].xCreate = tables::sqlite::xCreate;
+  modules[table_name].xConnect = tables::sqlite::xCreate;
+  modules[table_name].xBestIndex = tables::sqlite::xBestIndex;
+  modules[table_name].xDisconnect = tables::sqlite::xDestroy;
+  modules[table_name].xDestroy = tables::sqlite::xDestroy;
+  modules[table_name].xOpen = tables::sqlite::xOpen;
+  modules[table_name].xClose = tables::sqlite::xClose;
+  modules[table_name].xFilter = tables::sqlite::xFilter;
+  modules[table_name].xNext = tables::sqlite::xNext;
+  modules[table_name].xEof = tables::sqlite::xEof;
+  modules[table_name].xColumn = tables::sqlite::xColumn;
+  modules[table_name].xRowid = tables::sqlite::xRowid;
+
+  // Allow the table to receive INSERT/UPDATE/DROP events if it is
+  // implemented from an extension and is overwriting the right methods
+  // in the TablePlugin class
+  if (extension) {
+    std::lock_guard<std::mutex> locker(extension_table_list_mutex);
+    extension_table_list.push_back(table_name);
+
+    modules[table_name].xUpdate = tables::sqlite::xUpdate;
+  }
+
+  return &modules[table_name];
 }
-}
+} // namespace sqlite
+} // namespace tables
 
 Status attachTableInternal(const std::string& name,
                            const std::string& statement,
-                           const SQLiteDBInstanceRef& instance) {
+                           const SQLiteDBInstanceRef& instance,
+                           bool is_extension) {
   if (SQLiteDBManager::isDisabled(name)) {
     VLOG(1) << "Table " << name << " is disabled, not attaching";
     return Status(0, getStringForSQLiteReturnCode(0));
   }
 
-  // A static module structure does not need specific logic per-table.
-  // clang-format off
-  static sqlite3_module module = {
-      0,
-      tables::sqlite::xCreate,
-      tables::sqlite::xCreate,
-      tables::sqlite::xBestIndex,
-      tables::sqlite::xDestroy,
-      tables::sqlite::xDestroy,
-      tables::sqlite::xOpen,
-      tables::sqlite::xClose,
-      tables::sqlite::xFilter,
-      tables::sqlite::xNext,
-      tables::sqlite::xEof,
-      tables::sqlite::xColumn,
-      tables::sqlite::xRowid,
-      nullptr, /* Update */
-      nullptr, /* Begin */
-      nullptr, /* Sync */
-      nullptr, /* Commit */
-      nullptr, /* Rollback */
-      nullptr, /* FindFunction */
-      nullptr, /* Rename */
-      nullptr, /* Savepoint */
-      nullptr, /* Release */
-      nullptr, /* RollbackTo */
-  };
-  // clang-format on
+  sqlite3_module* module =
+      tables::sqlite::getVirtualTableModule(name, is_extension);
+  if (module == nullptr) {
+    auto error_message =
```

Comment:
I would just do this in the VLOG(1). In my opinion, the only time that returning an error message this verbose in a Status is useful is if you are checking it in code, or are going to print it out. Since you've already printed it out, you probably don't need to return it and `Status(1)` should suffice.

cc: @muffins Do you agree with this as a design decision? Please tell me I'm wrong if you think so.